Realm/RLMObjectStore.mm

```diff
@@ -182,7 +182,12 @@ void RLMAddObjectToRealm(__unsafe_unretained RLMObjectBase *const object,
 
     // get or create row
     bool created;
-    auto primaryGetter = [=](__unsafe_unretained RLMProperty *const p) { return [object valueForKey:p.getterName]; };
+    auto primaryGetter = [=](__unsafe_unretained RLMProperty *const prop) {
+        if (prop.swiftIvar) {
+            return static_cast<RLMOptionalBase *>(object_getIvar(object, prop.swiftIvar)).underlyingValue;
+        }
+        return [object valueForKey:prop.getterName];
```

Comment:
It seems that `[object valueForKey:prop.getterName]` should be `[object valueForKey:prop.name]`?